src/storage/v2/disk/storage.cpp

```diff
@@ -584,16 +651,13 @@ VerticesIterable DiskStorage::DiskAccessor::Vertices(LabelId label, PropertyId p
     /// TODO: couple this condition
     PropertyStore properties = utils::DeserializePropertiesFromLabelPropertyIndexStorage(it_value_str);
     auto prop_value = properties.GetProperty(property);
-    if (key_str.starts_with(utils::SerializeIdType(label) + "|" + utils::SerializeIdType(property)) &&
-        !utils::Contains(gids, curr_gid) && IsPropertyValueWithinInterval(prop_value, lower_bound, upper_bound)) {
+    if (key_str.starts_with(label_property_prefix) && !utils::Contains(gids, curr_gid) &&
```

Comment:
Also, I think you can do early return here in form of `continue` and then have one less indentation.